governance/tsc/tsc-2023-11-14.md

```diff
@@ -0,0 +1,190 @@
+* Location: https://meet.jit.si/Servo-TSC
+
+## Agenda
+
+* Status update
+* Dependencies mess
+* Servo crates maintenance
+* Multimedia status
+* Outreach
+* AOB
+
+## Notes
+
+Attending:
+* TSC members: cybai, Loirooriol, mrego, mrobinson, mukilan
+* Other: joshua-holmes, matlu, nicoburns, wusyong, zrhoffman
+
+### Introductions
+
+rego: As we have new people joining us today let's do a quick round of introductions.
+
+### Status update
+
+rego: Ongoing work on new layout engine. Floats are improving as time passes are more features are implemented. We recently upgraded SpiderMonkey, Stylo has been upgraded to March of this year, WebRender has been upgraded to an older version due to issues we still need to fix.
+
+rego: We have attended EclipseCon and participated in Oniro discussion there. We were talking about an experimental version of Servo for that platform. There are still a few issues, Mukilan can link to them. There has been progress. We did a presentation there.
+
+rego: https://github.com/eclipse-oniro4openharmony/manifest/issues/1
+
+rego: We have made improvements to the Minibrowser location bar. It supports figuring out the URL a bit better. We've been making progress on the API, looking at Tauri integration. We are exploring multi-WebView support. There were a bunch of issues a few weeks ago with dependency upgrades and with newer Python versions. Things are stable now though. Those are the main highlights.
+
+### Dependencies mess
+
+martin: The issue that we have with dependencies, is the number of dependencies we have and the effort it takes to upgrade them
+
+martin: because there are so many depedencies is easy for some package to depend on old versions; at some point we have to decide to don't do an upgrade or have two versions of the same dependency (which can increase the binary size)
+
+martin: I was wondering if we could come with some ideas, about how to avoid this problem in the future
+
+martin: that's why we had to fork some dependencies
+
+nico: is there a particular example of a dependency causing issues?
+
+martin: we were having issues with winit, that was depending on an old version of the bindings for MacOS; it was tricky to untangle the upgrade
+
+martin: winit depends on beta version of software that doesn't have stable release cycles
+
+martin: we have some dependencies that the major version is zero, so every major upgrade is a problem from us; quite a bit of API inestability in the rust ecosystem in general
+
+nico: one of the maintainers of winit is also the creator of one of these libraries, so it makes sense they're using new things
+
+nico: maybe using nightly rustc is causing issues?
+
+martin: we agree that this is causing us problem, one thing would be to make the changes needed to make servo build on a stable version of rust
+
+martin: it requires to stop using nightly only features
+
+nico: probably that's probably a big task
+
+nico: maybe you could do try to ...
```

Comment:
Sorry I missed this part while taking minutes yesterday, not sure if @mrobinson or @nicoburns remember this so we can fix the minutes. Thanks!

Reply:
I believe the suggestion was to try a specific environment variable that will let allow using feature flags in stable rust.

The variable name was not mentioned on the call, but based on my search, I found `[RUST_BOOTSTRAP](https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html)` that seems to have similar functionality, but its usage outside of rustc itself is [discouraged](https://internals.rust-lang.org/t/the-case-for-a-new-relese-channel-testing/14412).

Reply:
Thanks @mukilan!

```suggestion
nico: maybe you could do try a specific environment variable that will allow using feature flags in stable rust.
```